src/workerd/api/sockets.c++

```diff
@@ -78,7 +78,11 @@ jsg::Ref<Socket> connectImplNoOutputLock(
 }
 
 jsg::Ref<Socket> connectImpl(
-    jsg::Lock& js, kj::Maybe<jsg::Ref<Fetcher>> fetcher, kj::String address) {
+    jsg::Lock& js, kj::Maybe<jsg::Ref<Fetcher>> fetcher, kj::String address,
+    CompatibilityFlags::Reader featureFlags) {
+  if (!featureFlags.getTcpSocketsSupport()) {
```

Comment:
Instead of checking the flag at call time and throwing an exception, you should check the flag inside the `JSG_RESOURCE_TYPE` block and avoid registering the `connect` method at all if the flag isn't enabled.